Fix bugs: Allow missing variables for nullable args, accept string value for enum args.

timscott opened the pull request:
* Presence of variable was being required for nullable arg
* Was not handling enum args/variables.

timscott commented:
I'm not 100% confident I have fixed the enum issue in the correct way, although tests pass. Please take a close look at the change to `type_validation.cr`.

ziprandom commented:
Hey @timscott thanks for this pull request!

two things: 
1. [this](https://github.com/ziprandom/graphql-crystal/pull/20/files#diff-2565123e58dfa1077c0566847b12f16dR103) has a wrong name. The method implements `is_nonullable` ... is this method really necessary in the first place? couldn't you just make the type check directly in `schema_execute.cr`? 

2. could you provide a test to show what the problem is (was) with `enum` and change the title of the pull request to reflect the two things you fix here? 

apart from that your changes look good and I would happily merge them!

timscott commented:
1. I removed the `VariableDefinition#nullable?` convenience method. 

1. There was already a test for allowing string enum values, but it was giving a false positive because it was passing an enum, not a string. I fixed the test.
